src/Test/PdbUtilities/Shared/CustomDebugInfoReader.cs

```diff
@@ -533,6 +524,83 @@ public static ImmutableArray<string> GetVisualBasicImportStrings(this ISymUnmana
             }
         }
 
+        /// <summary>
+        /// If there dynamic locals or constants with SlotId == 0, check all locals and
+        /// constants with SlotId == 0 for duplicate names and discard duplicates since we
+        /// cannot determine which local or constant the dynamic info is associated with.
+        /// </summary>
+        private static ImmutableArray<DynamicLocalBucket> RemoveAmbiguousLocals(
+            ImmutableArray<DynamicLocalBucket> locals,
+            ArrayBuilder<ISymUnmanagedScope> scopes)
+        {
+            var localsAndConstants = PooledDictionary<string, object>.GetInstance();
+            var firstLocal = GetFirstLocal(scopes);
+            if (firstLocal != null)
+            {
+                localsAndConstants.Add(firstLocal.GetName(), firstLocal);
+            }
+            foreach (var scope in scopes)
+            {
+                foreach (var constant in scope.GetConstants())
+                {
+                    var name = constant.GetName();
+                    localsAndConstants[name] = localsAndConstants.ContainsKey(name) ? null : constant;
```

Comment:
```localsAndConstants[name]``` should be null if there are two or more of that name. I've updated the ```LocalDuplicateConstantAndConstantDynamic``` test to include more than two.